# Q1

In [1]:
import pandas as pd
import numpy as np
import scipy
from scipy import stats

In [2]:
cutlets = pd.read_csv("Cutlets.csv")
cutlets

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [3]:
#Shapiro-Wilk Test

from scipy.stats import shapiro
stats, p = shapiro(cutlets)
print('stats=%.3f, p=%.3f' % (stats,p)) # %.3f means .3 = round upto 3 digits, f=float no. 

stats=0.976, p=0.204


In [4]:
alpha = 0.05
if p>alpha:
    print("Fail to reject H0")
else:
    print("Reject H0")

Fail to reject H0


In [5]:
#Two-Tail Test

scipy.stats.ttest_ind(cutlets['Unit A'], cutlets['Unit B'])

Ttest_indResult(statistic=0.7228688704678061, pvalue=0.4722394724599501)

##### Since pvalue = 0.4722 > 0.05 
##### We do not reject Null Hypothesis

# Q2

In [13]:
import scipy.stats as stats
from scipy.stats import f_oneway

df2 = pd.read_csv("LabTAT.csv")
df2

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [14]:
#ANOVA Test
stats.f_oneway(df2.iloc[:,0],df2.iloc[:,1],df2.iloc[:,2],df2.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

#### Since pvalue = 2.115 < 0.05 
#### We accept Alternative Hypothesis

# Q3

In [23]:
#Since the dependent and independent variables are discrete we use Chi-Square Test
#H0 ->product sales ratio are same for males and females
#Ha ->product sales ratio are not same for males and females


from scipy.stats import chi2_contingency 
import scipy.stats as stats
data=[[50,142,131,70],[550,351,480,350]]
m=chi2_contingency(data)
m

(80.27295426602495,
 2.682172557281901e-17,
 3,
 array([[111.01694915,  91.21892655, 113.05225989,  77.71186441],
        [488.98305085, 401.78107345, 497.94774011, 342.28813559]]))

##### Since pvalue = 2.68 < 0.05 
##### Reject Null Hypothesis

## Q4

In [24]:
df4 = pd.read_csv("Costomer+OrderForm.csv")
df4

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [25]:
m1=df4['Phillippines'].value_counts()
m2=df4['Indonesia'].value_counts()
m3=df4['Malta'].value_counts()
m4=df4['India'].value_counts()
frame = [m1,m2,m3,m4]
result = pd.concat(frame, axis=1)
from scipy.stats import chi2_contingency
data1=[[result.iloc[0,0:4]],[result.iloc[1,0:4]]]
res1 = chi2_contingency(data1)
res1

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[[271.75, 271.75, 271.75, 271.75]],
 
        [[ 28.25,  28.25,  28.25,  28.25]]]))

##### Since pvalue = 0.277 > 0.05 
##### Do not reject Null Hypothesis

## Q5

In [41]:
faltoons = pd.read_csv("Faltoons.csv")
faltoons

Weekdays Weekend
0       Male  Female
1     Female    Male
2     Female    Male
3       Male  Female
4     Female  Female
..       ...     ...
395   Female    Male
396   Female  Female
397   Female  Female
398   Female    Male
399   Female  Female

[400 rows x 2 columns]

In [42]:
faltoons["Weekdays"].value_counts()

Female    287
Male      113
Name: Weekdays, dtype: int64

In [43]:
cleanup = {"Weekdays":{"Female":1,"Male":0},
           "Weekend":{"Female":1,"Male":0}
}

In [44]:
faltoons.replace(cleanup, inplace=True)

In [45]:
faltoons

Weekdays  Weekend
0           0        1
1           1        0
2           1        0
3           0        1
4           1        1
..        ...      ...
395         1        0
396         1        1
397         1        1
398         1        0
399         1        1

[400 rows x 2 columns]

In [47]:
faltoons["Weekdays"].mean()

0.7175

In [48]:
faltoons["Weekend"].mean()

0.5825

In [49]:
import statsmodels.api as sm
sm.stats.ttest_ind(faltoons["Weekdays"], faltoons["Weekend"])

(4.038383581687957, 5.9013879428528154e-05, 798.0)

##### Since pvalue = 5.901 < 0.05
##### Reject Null hypothesis i.e. all proportion are equal